src/librustc/ty/layout.rs

```diff
@@ -1646,6 +1691,14 @@ impl<'a, 'tcx, C> TyLayoutMethods<'tcx, C> for Ty<'tcx>
 
     fn field(this: TyLayout<'tcx>, cx: &C, i: usize) -> C::TyLayout {
         let tcx = cx.tcx();
+        let handle_discriminant = |discr: &Scalar| -> C::TyLayout {
```

Comment:
I'd call this `make_discr_field` (which we should maybe pre-intern, this could be unnecessarily expensive).